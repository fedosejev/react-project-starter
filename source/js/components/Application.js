import React, { Component } from 'react';
import { func, object } from 'prop-types';

class Application extends Component {
  doSomething = () => {
    const {
      doSomething,
    } = this.props;

    doSomething();
  }

  render() {
    const {
      application,
    } = this.props;

    const {
      isDoingSomething,
    } = application;

    const buttonLabel = isDoingSomething ? 'Doing something' : 'Do something';

    return (
      <div className="container">
        <div className="row">
          <div className="col-sm-12">
            <h1>React.js Application</h1>
            <button onClick={this.doSomething} className="btn btn-default">{buttonLabel}</button>
          </div>
        </div>
      </div>
    );
  }
}

Application.propTypes = {
  application: object.isRequired,
  doSomething: func.isRequired,
};

export default Application;
