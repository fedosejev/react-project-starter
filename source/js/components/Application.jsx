import React from 'react';

let Application = React.createClass({
  render: function () {
    return (
      <div className="container">
        <div className="row">
          <div className="col-sm-12">
            <h1>React.js Application</h1>
          </div>
        </div>
      </div>
    );
  }
});

export default Application;
