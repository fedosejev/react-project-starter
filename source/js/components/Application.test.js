import React from 'react';
import renderer from 'react-test-renderer';
import Application from './Application';

const mockFunction = jest.fn();
const mockApplicationState = {
  isDoingSomething: false,
};

describe('Ad', () => {
  test('SHOULD render correctly', () => {
    const tree = renderer.create(
      <Application
        application={mockApplicationState}
        doSomething={mockFunction}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
