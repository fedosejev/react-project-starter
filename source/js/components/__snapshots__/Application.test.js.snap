// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ad SHOULD render correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-sm-12"
    >
      <h1>
        React.js Application
      </h1>
      <button
        className="btn btn-default"
        onClick={[Function]}
      >
        Do something
      </button>
    </div>
  </div>
</div>
`;
